data_scrapers/sonoma_county.py

```diff
@@ -0,0 +1,216 @@
+    rows = get_rows(race_eth_tag)
+    for row in rows:
+        group_name, cases, _pct = get_cells(row)
+        if group_name not in race_transform:
+            raise FutureWarning('The racial group {0} is new in the data -- please adjust the scraper accordingly')
+        internal_name = race_transform[group_name]
+        race_cases[internal_name] = int(cases)
+    race_cases['Unknown'] = get_unknown_race(race_eth_tag)
+    return race_cases
+
+def transform_total_hospitalizations(hospital_tag: element.Tag) -> Dict[str, int]:
+    """
+    Takes in a BeautifulSoup tag of the cases by hospitalization table and
+    returns a dictionary with the numbers of hospitalized and non-hospitalized
+    cases
+    """
+    hospitalizations = {}
+    rows = get_rows(hospital_tag)
+    for row in rows:
+        hospitalized, number, _pct = get_cells(row)
+        if hospitalized == 'Yes':
+            hospitalizations['hospitalized'] = int(number)
+        else:
+            hospitalizations['not_hospitalized'] = int(number)
+    return hospitalizations
+
+def transform_gender_hospitalizations(hospital_tag: element.Tag) -> Dict[str, float]:
+    """
+    Takes in a BeautifulSoup tag representing the percent of cases hospitalized
+    by gender and returns a dictionary of those percentages in float form
+    e.g. 9% is 0.09
+    """
+    hospitalized = {}
+    rows = get_rows(hospital_tag)
+    for row in rows:
+        gender, no, yes = get_cells(row)
+        yes_int = int(yes.replace('%', ''))
+        hospitalized[gender] = (yes_int / 100)
+    return hospitalized
+
+def get_county() -> Dict:
+    """Main method for populating county data .json"""
+    url = 'https://socoemergency.org/emergency/novel-coronavirus/coronavirus-cases/'
+    page = requests.get(url)
+    sonoma_soup = BeautifulSoup(page.content, 'html.parser')
+    tables = sonoma_soup.findAll('table')[4:] # we don't need the first three tables
+
+    try:
+        # we have a lot more data here than we are using
+        hist_cases, cases_by_source, cases_by_race, total_tests, cases_by_region, region_guide, hospitalized, underlying_cond, symptoms, cases_by_gender, underlying_cond_by_gender, hospitalized_by_gender, symptoms_female, symptoms_male, symptoms_desc, cases_by_age, symptoms_by_age, underlying_cond_by_age = tables
+    except ValueError:
+        raise FutureWarning('The number of values on the page has changed -- please adjust the scraper')
+
+    model = {
+        'name': 'Sonoma County',
+        'update_time': generate_update_time(sonoma_soup),
+        'source': url,
+        'meta_from_source': get_source_meta(sonoma_soup),
+        'meta_from_baypd': 'Racial "Other" category includes "Black/African American, American Indian/Alaska Native, and Other"',
+        'series': transform_cases(hist_cases),
+        'case_totals': {
+            'transmission_cat': transform_transmission(cases_by_source),
+            'age_group': generic_transform(cases_by_age),
```

Comment:
This gets us the wrong format:

```js
        "age_group": {
            "0-17": 58,
            "18-49": 190,
            "50-64": 78,
            "65 and Above": 47,
            "Under Investigation": 0
        },
```

But we should have a list, and the group names should be slightly different: https://github.com/sfbrigade/data-covid19-sfbayarea/tree/master/data_models#cases-2

```js
[
	{"group": "0_to_17", "raw_count": -1 },
	{"group": "18_to_49", "raw_count": -1 },
	{"group": "50_to_64", "raw_count": -1 },
	{"group": "65_and_older", "raw_count": -1 },
	{"group": "Unknown", "raw_count": -1 }
]
```

Reply:
Our data_model.json had inconsistent use of "age_group" values (only 1 instance was a list). I changed all the "age_group" tables to be lists in [#52, in progress](https://github.com/sfbrigade/data-covid19-sfbayarea/pull/52). 
As far as using the group names, I realized I need to go do that for Alameda County and SF county. And we will probably have to manually order the list by group-name, since they're often coming through as unordered .json objects (in curly braces).